tutorial7_state_management/demo/Count.js

```diff
@@ -0,0 +1,55 @@
+class Count {
+
+  constructor(state, setGlobalState) {
+    this.container = d3.select("#count")
+    this.duration = 1000
+  }
+
+  draw(state, setGlobalState) {
+    console.log("now I am drawing my count");
+
+    const filteredData = state.data.find(d => state.selectedState === d.State);
+    const metrics = ["Age < 20", "Age 20-65", "Age 65+"];
+    const metricData = metrics.map(metric => {
+      return {
+        state: state.selectedState,
+        metric: metric,
+        value: filteredData ? filteredData[metric] : 0,
+      };
+    });
```

Comment:
I had another thought that we could do this filter in `init` and then pass both `barchart` and `count` the filteredData instead of all of state. 

Not sold on that but that way we can avoid the duplicated logic here. 
Then again maybe this way is cleaner, that each component has it's own logic on state.